Refinements

ettore-infn opened the pull request:
Fixes and adds small details talked about in the meetings

ettore-infn committed:
Implement config/settings file. Added a config file where all the env variables are exported. Changed the env variables call in every file

ettore-infn committed:
added ADMIN group role that can perform all the actions of site admin, site tester and sla manager

ettore-infn committed:
fixed errors handling and message/toaster showing

ettore-infn committed:
small fixes to improve UX & UI

ettore-infn committed:
added "Unassign provider" button

jacogasp commented on src/app/api/ssr/providers/route.ts:
```diff
@@ -26,14 +27,14 @@ export async function GET() {
   const userRoles = await findUserRoles();
 
-  let apiUrl = `${process.env.API_SERVER_URL}/providers?&status=ready&status=evaluation&status=pre-production&status=active&status=deprecated&status=removed&status=degraded&status=maintenance&status=re-evaluation`;
+  let apiUrl = `${settings.apiServerUrl}/providers?&status=ready&status=evaluation&status=pre-production&status=active&status=deprecated&status=removed&status=degraded&status=maintenance&status=re-evaluation`;
```
Usa https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams o semplicemente (e più veloce) tieni la stringa ma formattala meglio, non si capisce nulla di quali e quanti parametri ci sono

ettore-infn commented on src/app/api/ssr/providers/[id]/testers/route.ts:
```diff
@@ -15,17 +16,14 @@ export async function POST(req: Request, { params }: Params) {
 
   const body = await req.json();
```
Risolto passando nel body esclusivamente il dato richiesto (id)

ettore-infn commented on src/app/api/ssr/providers/route.ts:
```diff
@@ -10,7 +11,7 @@ export async function POST(req: Request) {
   }
 
   const body = await req.json();
```
Risolto passando esclusivamente i parametri necessari a completare la chiamata

ettore-infn committed:
Fix suggested in the PR review:
- Fix typo and code refactor
- Refactor API routes to use a centralized IDPS URL
- Fix error string to prevent passing sensitive data
- Sanitize POST requests payload by explicitly mapping allowed fields
